pkg/common/cns-lib/volume/manager.go

```diff
@@ -1114,6 +1114,14 @@ func (m *defaultManager) deleteVolume(ctx context.Context, volumeID string, dele
 	}
 	volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 	if volumeOperationRes.Fault != nil {
+		// If volume is not found on host, then is present in CNS DB, we will get NotFound fault.
+		// Send back success as the volume is already deleted.
+		_, isNotFoundFault := volumeOperationRes.Fault.Fault.(*vim25types.NotFound)
```

Comment:
Can we make use of `ExtractFaultTypeFromVolumeResponseResult` to extract the fault type and then compare it with the expected fault type?
if `Fault.Fault` is nil, we will run into a panic here.

Reply:
done